nixos/modules/services/audio/ympd.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ympd;
+in {
+
+  ###### interface
+
+  options = {
+
+    services.ympd = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable ympd, the MPD Web GUI.";
+      };
+
+      webPort = mkOption {
+        default = 8080;
+        description = "The port where ympd's web interface will be available.";
+      };
+
+      user = mkOption {
+        default = "ympd";
+        description = "User account under which ympd runs.";
+      };
+
+      group = mkOption {
+        default = "ympd";
+        description = "Group to assign to ympd user.";
+      };
+
+      mpd = {
+        host = mkOption {
+          default = "localhost";
+          description = "The host where MPD is listening.";
+        };
+
+        port = mkOption {
+          default = 6600;
+          description = "The port where MPD is listening.";
+        };
+      };
+
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.services.ympd = {
+      after = [ "sound.target" "mpd.service" ];
+      description = "Standalone MPD Web GUI written in C";
+      wantedBy = [ "multi-user.target" ];
+      path = [ pkgs.ympd ];
+      script = "exec ympd --host ${cfg.mpd.host} --port ${toString cfg.mpd.port} --webport ${toString cfg.webPort} --user ${cfg.user}";
```

Comment:
This generates a shell script that will be executed. You can use `serviceConfig.ExecStart` and drop the exec for simplification. If you use `"${pkgs.ympd}/bin/ympd"` here you can also drop the `path`.